File: LeetCode/MoveZeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        i = 0
        length = len(nums)
        zeros = 0
        while i < length:
            if nums[i] == 0:
                del nums[i]
                zeros += 1
                length -= 1
            else:
                i += 1
        nums += [0]*zeros

File: LeetCode/MoveZeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        int nonzero_i = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] != 0) {
                nums[nonzero_i++] = nums[i];
            }
        }
        for (int i = nonzero_i; i < n; i++) {
            nums[i] = 0;
        }
    }
};
